src/main/java/duke/TaskData.java

```diff
@@ -0,0 +1,100 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.task.*;
+
+public class TaskData {
+    public static void storeTaskData(File taskDataFile, ArrayList<Task> tasks) throws IOException {
+        FileWriter fileWriter = new FileWriter(taskDataFile);
+        if(!taskDataFile.exists()){
+            taskDataFile.createNewFile();
+        }
+        for(Task task : tasks){
+            switch (task.getTaskType()){
+            case TODO:
+                fileWriter.write(encodeTaskNameAndStatus(task, CommandInputs.ADD_TODO_COMMAND_INPUT) + System.lineSeparator());
+                break;
+            case DEADLINE:
+                fileWriter.write(encodeDeadlineData((Deadline)task) + System.lineSeparator());
+                break;
+            case EVENT:
+                fileWriter.write(encodeEventData((Event)task) + System.lineSeparator());
+                break;
+            default:
+                // throw some error
```

Comment:
Consider removing this comment if you are not planning to implement anything more here